python/sglang/srt/utils.py

```diff
@@ -138,30 +141,48 @@ def is_port_available(port):
             return False
 
 
+def alloc_usable_network_port(num, used_list=()):
+    port_list = []
+    for port in range(10000, 65536):
+        if port in used_list:
+            continue
+        if is_port_available(port):
+            port_list.append(port)
+        if len(port_list) == num:
+            return port_list
+    return port_list
+
+
 def allocate_init_ports(
     port: Optional[int] = None,
     additional_ports: Optional[List[int]] = None,
     tp_size: int = 1,
+    dp_size: int = 1,
 ):
-    if additional_ports:
-        ret_ports = [port] + additional_ports
-    else:
-        ret_ports = [port]
-
-    ret_ports = list(set(x for x in ret_ports if is_port_available(x)))
-    cur_port = ret_ports[-1] + 1 if len(ret_ports) > 0 else 10000
-
-    while len(ret_ports) < 5 + tp_size:
-        if cur_port not in ret_ports and is_port_available(cur_port):
-            ret_ports.append(cur_port)
-        cur_port += 1
+    ret_ports = []
```

Comment:
Use this new logic https://github.com/sgl-project/sglang/blob/55c16436273d4a42f7cfe342df5f10ad05a8d0fe/python/sglang/srt/utils.py#L141-L164